templates/extensions/foundry/packages/foundry/script/generateTsAbis.js

```diff
@@ -36,12 +36,14 @@ function getArtifactOfContract(contractName) {
 
 function getInheritedFromContracts(artifact) {
   let inheritedFromContracts = [];
-  for (const astNode of artifact.ast.nodes) {
-    if (astNode.nodeType == "ContractDefinition") {
-      if (astNode.baseContracts.length > 0) {
-        inheritedFromContracts = astNode.baseContracts.map(
-          ({ baseName }) => baseName.name
-        );
+  if('ast' in artifact) {
```

Comment:
Thanks for your pr!

`'ast' in artifact` is true even when `ast` is undefined.

<img width="273" alt="image" src="https://github.com/scaffold-eth/scaffold-eth-2/assets/25638585/47f37ce6-9329-4c68-b8d1-4f5609c18398">

so it should fail even in your case

Better to use 
`if (artifact?.ast) {`

Reply:
@dragonsisdumb could you please update it ?